delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/batch/BatchCopy.scala

```diff
@@ -112,14 +114,23 @@ object BatchCopy {
     private def notEnoughSpace(totalSize: Long, spaceLeft: Long, destStorage: Iri) =
       IO.raiseError(TotalCopySizeTooLarge(totalSize, spaceLeft, destStorage))
 
-    private def fetchFileAndValidateStorage(id: FileId)(implicit c: Caller): IO[(File, Storage)] = {
+    private def fetchFileAndValidateStorage(id: FileId)(implicit c: Caller): IO[(File, Storage)] =
```

Comment:
You should be able to use `ResourceRef` directly here, I see that `FileId` use `IdSegmentRef`
Those are meant to be used for segment in the api enpoint where the `Iri` can be compacted.
The source file being in a payload, you should be able to use `ResourceRef` directly and make things simpler